doc/source/ray-logging.rst

```diff
@@ -140,3 +140,38 @@ If you'd like to change the log rotation configuration, you can do it by specify
     RAY_ROTATION_MAX_BYTES=1024; ray start --head # Start a ray instance with maxBytes 1KB.
     RAY_ROTATION_BACKUP_COUNT=1; ray start --head # Start a ray instance with backupCount 1.
 
+Redirecting Ray logs to stderr
```

Comment:
Do we have a unit test for this?

Reply:
There's an existing unit test that I extended that ensures the log files are not created; however, they do not ensure that log records are emitted via stderr.

Reply:
Can you give me a pointer? At a high level, I would expect a test that runs the same code before / after RAY_LOG_TO_STDERR is set, and checks the right things appear in stderr with both settings.

Reply:
We could pick the startup log record from each component and ensure that each is emitted via stderr, that's essentially what I did when manually testing that this worked.

Reply:
@ericl [This](https://github.com/ray-project/ray/pull/21767/files#diff-c483a882c837623be3869968fa21a8cbce2f5f042fcd5179bbc36bcf2a7edeff) is the test that I extended, that was the existing test for `GLOG_logtostderr`. I agree on this not being proper coverage, since an implementation that just drops all log records wouldn't caused the test to fail.

I'm aware of the `capsys` fixture that we can use. Do we have any existing logging tests that verify log record content on disk?

Reply:
There's run_driver_as_string which is commonly used for complete env setting / capture.

Reply:
Covered all except for the Ray Client server, have a TODO for that one.